### PR TITLE
[Target] Fix device mask issue and typos

### PR DESCRIPTION
**Pull request by leeexyz**
Hi Community,

Currently, I am working on the Heterogeneous Execution by using `Composite Target`. I found it is not included in the `Device.STR2MASK` map and will be failed when compiling with it. So, I added it back and checked all the other targets. Not surprisingly, there are some other issues. Now, I add a `Consistency verification case` to help us check it. :)

In addition to these, I still have some doubts.
1.  Since introducing the new `Target System`,  the device type is saved in each Target (`tvm.target.Target.kind`). It seems that some attributes of the Device in `runtime_ctypes.py` are redundant, such as `MASK2STR` and `STR2MASK`. Then the issue raised here is the same or equaled features located in different locations make things messy. I think we can refactor the `Runtime Device`.
2. It seems that the devices of `Composite Target` cannot be optimized or even recognized by all relay passes. Ideally, the `Partition Pass` can perform the partitioning and annotation automatically based on those devices (order does matter), and then select the best op strategy. So my question is: What is the current status for `Composite Target` and is anyone working on it?

Any help would be much appreciated. :)

### CODE REVIEW

File: python/tvm/_ffi/runtime_ctypes.py [hunk 1]
```diff
@@ -217,13 +216,15 @@ class Device(ctypes.Structure):
         "stackvm": 1,
         "cpu": 1,
         "c": 1,
+        "hybrid": 1,
```
Comment (masahi) [inline python/tvm/_ffi/runtime_ctypes.py:219]: what is hybrid?

**Comment by masahi** [comment]
cc @zxybazh @junrushao1994

File: python/tvm/_ffi/runtime_ctypes.py (hunk 1 above)
Comment (leeexyz) [reply python/tvm/_ffi/runtime_ctypes.py:219]: Not sure about it. There is no comment or RFC.
https://github.com/apache/tvm/blob/main/src/target/target_kind.cc#L400-L401

Comment (junrushao1994) [reply python/tvm/_ffi/runtime_ctypes.py:219]: @masahi there is a legacy thing called TE hybrid script. Probably it’s related?

**Review by masahi — APPROVED** [review]

**Pull request merged**